validator/std_validator.go

```diff
@@ -0,0 +1,70 @@
+package validator
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// Validate validates the cobra commands
+// and all the descendants
+func Validate(cmd *cobra.Command) error {
+	err := handleDefaultErrors(cmd)
+	if err != nil {
+		return err
+	}
+
+	// handeling errors for subcommands of a command
+	subCommmands := cmd.Commands()
+	for _, child := range subCommmands {
+		err := handleDefaultErrors(child)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ValidateCustom will allow user to
```

Comment:
This is wrong. Every rule should be custom rule. Create interface for rule with single method that validates single command and then traverse thru nested structure.  This is easier than current architecture and will allow anyone to drop any rules - users will always need to define rules somehow or override them, but it should not require to actually write them - for example fieldExist or string length rule needs to have some arguments to be defined.